treeherder/intermittents_commenter/commenter.py

```diff
@@ -297,46 +309,94 @@ def get_bug_stats(self, startday, endday):
             .filter(total__gte=threshold)
             .values_list("bug_id", flat=True)
         )
-
         bugs = (
             BugJobMap.failures.by_date(startday, endday)
             .filter(bug_id__in=bug_ids)
+            .order_by("job__machine_platform__platform")
             .values(
                 "job__repository__name",
                 "job__machine_platform__platform",
                 "bug_id",
                 "job__option_collection_hash",
+                "job__signature__job_type_name",
+                "job__signature__job_group_name",
             )
         )
-
         option_collection_map = OptionCollection.objects.get_option_collection_map()
-        bug_map = dict()
+        bug_map = self.build_bug_map(bugs, option_collection_map)
+        return bug_map, bug_ids
 
+    def fetch_test_variants(self):
+        mozilla_central_url = "https://hg.mozilla.org/mozilla-central"
+        variant_file_url = f"{mozilla_central_url}/raw-file/tip/taskcluster/kinds/test/variants.yml"
+        response = requests.get(variant_file_url, headers={"User-agent": "mach-test-info/1.0"})
+        return yaml.safe_load(response.text)
+
+    def get_test_variant(self, test_suite):
+        test_variants = self.fetch_test_variants()
+        # iterate through variants, allow for Base-[variant_list]
+        variant_symbols = sorted(
+            [
+                test_variants[v]["suffix"]
+                for v in test_variants
+                if test_variants[v].get("suffix", "")
+            ],
+            key=len,
+            reverse=True,
+        )
+        # strip known variants
+        # build a list of known variants
+        base_symbol = test_suite
+        for variant in variant_symbols:
+            if f"-{variant}" in base_symbol:
+                return variant
+        return "no variant"
```

Comment:
the only thing here is we can have combination variants, like v1+v2, it will look like `test-v1-v2` in the test_suite.  in this case, I recommend instead of `return variant`:
```
found_variants = []
for variant in variant_symbols:
  found_variants.append(variant)
  base_symbol = base_symbol.replace(f"-{variant}", "")
if not found_variants:
  found_variants = "no_variant"
return '-'.join(found_variants)
```

Reply:
I try with on the fly job name modification which seems lighter to me. But I can see about adding an extra file in sample data if you think it's better.